### PR TITLE
Add layout

### PR DESCRIPTION
**Pull request by OriGuyUniqueDev**
- added basic layout without styles header, main, and footer
- added login page
- added register page
- added basic routing: /home (layout) / (login page) /register (register page)

### CODE REVIEW

File: client/src/components/layout/Layout.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import Header from "./Header/Header";
+import Main from "./Main/Main";
+
+interface LayoutProps {}
+
+const Layout: FunctionComponent<LayoutProps> = () => {
+	return (
+		<Box sx={{ display: "flex", flexDirection: "column", minHeight: "100vh" }}>
+			<Header></Header>
+			<Main></Main>
```
Comment (bk-software) [inline client/src/components/layout/Layout.tsx:13]: Layout and Main should have children right?

File: client/src/components/signin_register/Signin.tsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+			>
+				HackerU
+			</Link>{" "}
+			{new Date().getFullYear()}
+			{"."}
+		</Typography>
+	);
+}
+
+const theme = createTheme();
```
Comment (bk-software) [inline client/src/components/signin_register/Signin.tsx:38]: remove the theme from here

**Review by bk-software — CHANGES_REQUESTED** [review]

File: client/src/components/layout/Layout.tsx (hunk 1 above)
Comment (OriGuyUniqueDev) [reply client/src/components/layout/Layout.tsx]: i fixed it

File: client/src/components/signin_register/Signin.tsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+				sx={{ height: "100vh" }}
+			>
+				<CssBaseline />
+				<Grid
+					item
+					xs={false}
+					sm={4}
+					md={7}
+					sx={{
+						backgroundImage: "url(https://source.unsplash.com/random)",
```
[the remark this answers is not in the document]
Comment (OriGuyUniqueDev) [reply client/src/components/signin_register/Signin.tsx:64]: In my vscode, it looks fine.
What can we do?

File: client/src/components/signin_register/Signin.tsx (hunk 2 above)
Comment (OriGuyUniqueDev) [reply client/src/components/signin_register/Signin.tsx:38]: I removed it

File: client/src/main.tsx [hunk 4]
```diff
@@ -1,10 +1,29 @@
 import React from 'react'
 import ReactDOM from 'react-dom/client'
+import {createBrowserRouter,RouterProvider,} from "react-router-dom";
 import App from './App'
+import Layout from './components/layout/Layout';
+import Register from './components/signin_register/Register';
+import SignIn from './components/signin_register/Signin';
 import './index.css'
 
+const router = createBrowserRouter([
```
[the remark this answers is not in the document]
Comment (OriGuyUniqueDev) [reply client/src/main.tsx:11]: I did it

File: client/src/components/signin_register/Signin.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (bk-software) [reply client/src/components/signin_register/Signin.tsx]: Did you try to ask google about it?

File: client/src/components/signin_register/Signin.tsx [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+import Paper from "@mui/material/Paper";
+import Box from "@mui/material/Box";
+import Grid from "@mui/material/Grid";
+import LockOutlinedIcon from "@mui/icons-material/LockOutlined";
+import Typography from "@mui/material/Typography";
+import { createTheme, ThemeProvider } from "@mui/material/styles";
+import { Link as RouterLink, LinkProps as RouterLinkProps } from "react-router-dom";
+import { Container } from "@mui/material";
+
+function Copyright(props: any) {
```
Comment (bk-software) [inline client/src/components/signin_register/Signin.tsx:18]: It is better that this will move to another component. Right?

**Pull request merged**